### PR TITLE
Prepare to reconfigure ESLint

### PR DESCRIPTION
**Pull request by kachkaev**
Precursor to #508

- Slightly cleanup `.eslinrc.js`

- Bump ESLint deps to avoid this warning:

```
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.7.0

YOUR TYPESCRIPT VERSION: 4.7.4

Please only submit bug reports when using the officially supported version.

=============
```

ESLint was last bumped in Feb 2022 via https://github.com/blockprotocol/blockprotocol/pull/209. TypeScript ESLint plugin was last bumped in April 2022 via https://github.com/blockprotocol/blockprotocol/pull/298.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,14 +12,14 @@
     "fix": "npm-run-all --continue-on-error \"fix:*\"",
     "@todo.1": "Remove the ignore argument below once react-block-loader is upgraded to 0.2 spec",
     "fix:dependency-version-consistency": "check-dependency-version-consistency --fix . --ignore-package=react-block-loader",
-    "fix:eslint": "eslint --report-unused-disable-directives \"**/*\" --fix",
+    "fix:eslint": "eslint \"**/*\" --fix",
```
Comment (kachkaev) [inline package.json:15]: Extracted to `reportUnusedDisableDirectives: true`

File: packages/create-block-app/create-block-app.js [hunk 2]
```diff
@@ -2,13 +2,13 @@
 
 /* eslint-disable no-console */
 
-const { promisify } = require("util");
-const child_process = require("child_process");
+const { promisify } = require("node:util");
```
Comment (kachkaev) [inline packages/create-block-app/create-block-app.js:5]: Result of `yarn fix:eslint` (must be a new recommended rule)

File: .eslintrc.cjs [hunk 3]
```diff
@@ -15,9 +15,11 @@ const sharedNoRestrictedImportsConfig = {
   ],
 };
 
+/** @type {import("eslint").Linter.Config} */
 module.exports = {
   // this is the highest config lower ones will automatically extend
   root: true,
+  reportUnusedDisableDirectives: true,
```
Comment (kachkaev) [inline .eslintrc.cjs:22]: Should work in editors unlike `--report-unused-disable-directives` package.json scripts

File: .eslintrc.cjs [hunk 4]
```diff
@@ -380,12 +376,6 @@ module.exports = {
         ],
       },
     },
-    {
-      files: ["site/**"],
-      rules: {
-        "unicorn/filename-case": "error",
-      },
-    },
```
Comment (kachkaev) [inline .eslintrc.cjs:388]: I forgot to remove this in https://github.com/blockprotocol/blockprotocol/pull/305

File: .eslintrc.cjs [hunk 5]
```diff
@@ -15,9 +15,11 @@ const sharedNoRestrictedImportsConfig = {
   ],
 };
 
+/** @type {import("eslint").Linter.Config} */
```
Comment (yusufkinatas) [inline .eslintrc.cjs:18]: Great for DX, liked it!

**Review by yusufkinatas — APPROVED** [review]
Tested on my local & confirmed it works.

**Pull request merged**